control-plane/agents/core/src/core/reconciler/poller.rs

```diff
@@ -28,6 +28,7 @@ impl ReconcilerWorker {
             Box::new(nexus::NexusReconciler::new()),
             Box::new(volume::VolumeReconciler::new()),
             Box::new(PersistentStoreReconciler::new()),
+            Box::new(replica::garbage_collector::GarbageCollector::new()),
```

Comment:
just a suggestion, to keep in line with the existing nexus and volumes, the garbage_collector could be made private within the replica as ReplicaReconciler, which in turns calls its own garbage collector